### DISCUSSION TITLE
Is there a way to connect DLLs to an AIR game for PC/Windows?

### DISCUSSION THREAD

**Discussion by GeneralVimes** [airsdk/Adobe-Runtime-Support#1145] [Q&A] [answered]
I'm building a turn-based strategy using AIR/Starling. It will need strong AI. I want to let the players write their own AI and connect to the game.

I remember, 11 years ago, when I was making my another TBS, Cards Wars, on Object Pascal in Lazarus, there was a possibility to move the service functions to DLL files and let the players create own DLL files with their AIs. It was fun to organize competitions between different AIs.

How is the DLL connection thing in AIR. Is it possible, and did anyone have experience with it?

**Comment by itlancer** [comment, marked as answer]
It is possible by writing Native Extension for AIR.
I think you should start from here: http://easynativeextensions.com/windows-tutorial-the-native-dll-project/

Also via C++ you can use dynamic DLL loading (or call separate functions from DLL): https://stackoverflow.com/questions/8696653/dynamically-load-a-function-from-a-dll
